packages/applications/ssr/src/components/molecules/CommunePicker.tsx

```diff
@@ -0,0 +1,122 @@
+'use client';
+
+import React, { useState, useEffect } from 'react';
+import Autocomplete from '@mui/material/Autocomplete';
+import Input, { InputProps } from '@codegouvfr/react-dsfr/Input';
+import { debounce } from '@mui/material/utils';
+
+type GeoApiCommuneItem = {
+  nom: string;
+  codesPostaux: string[];
+  departement: {
+    code: string;
+    nom: string;
+  };
+  region: {
+    code: string;
+    nom: string;
+  };
+};
+
+type Commune = {
+  commune: string;
+  codePostal: string;
+  departement: string;
+  region: string;
+};
+
+const fetchCommunes = async (search: string, signal?: AbortSignal): Promise<Commune[]> => {
+  const params = new URLSearchParams({
+    nom: search,
+    fields: 'departement,region,codesPostaux',
+    boost: 'population',
+    limit: '10',
+  });
+
+  const response = await fetch(`https://geo.api.gouv.fr/communes?${params}`, { signal });
+  const data: GeoApiCommuneItem[] = await response.json();
+  return data.map(({ nom, region, departement, codesPostaux }) => ({
+    commune: nom,
+    region: region.nom,
+    departement: departement.nom,
+    codePostal: codesPostaux[0],
+  }));
+};
+
+export type CommunePickerProps = {
+  onSelected?: (commune: Commune | null) => void;
+  defaultValue?: Commune;
+} & Pick<InputProps, 'className' | 'label' | 'nativeInputProps'>;
+
+export const CommunePicker: React.FC<CommunePickerProps> = ({
+  label,
+  nativeInputProps,
+  defaultValue,
+  className,
+  onSelected,
+}) => {
+  const [communes, setCommunes] = useState<Commune[]>([]);
+  const [selectedCommune, setSelectedCommune] = useState<Commune | null>(defaultValue ?? null);
+  const [search, setSearch] = useState('');
+  const [loading, setLoading] = useState(false);
+
+  useEffect(() => {
```

Comment:
Je ne suis pas sûre qu'il y ait besoin d'un useeffect et d'un state ici, tu peux gérer plus simplement je crois

```suggestion
  const [communes, setCommunes] = useState<Commune[]>([]);
  const [selectedCommune, setSelectedCommune] = useState<Commune | null>(defaultValue ?? null);
  const [loading, setLoading] = useState(false);

  const handleSearch = async (searchValue: string) => {
    if (!searchValue) {
      setCommunes([]);
      setLoading(false);
      return;
    }

    setLoading(true);
    try {
      const controller = new AbortController();
      const data = await fetchCommunes(searchValue, controller.signal);
      setCommunes(data);
    } catch (error) {
      console.error('Error fetching communes:', error);
    } finally {
      setLoading(false);
    }
  };

  const searchDelayed = debounce(handleSearch, 400);```

Reply:
ça évite les effets de bord des useeffect